mirage/xen-arm-builder
# Optionally include newer versions of Linaro/Ubuntu/Xen/Linux

This doesn't change the default build unless you set `DISTROVER=vivid`. The differences are:
- Linaro 15.06 root filesystem
  - Includes a prebuilt Xen 4.5 binary (fixes #52), so for now the Xen build is skipped.
- Ubuntu 15.04 (Vivid Vervet)
  - GCC 4.9.2
- Linux stable 4.1.y mainline
  - Config based on config-cubie2-docker
- NTP daemon (fixes #48)
- Some changes to make it easier to use a caching proxy for apt while building the image

Tested with:
- Ran `make-linux-guest.sh` and guest booted OK. Minor changes:
  - `xenbr0` -> `br0` (no XAPI yet)
  - trusty -> vivid (libpam-systemd post-install failed for trusty guest)
- `opam init`
- `opam switch set 4.02.1`
  - I tried 4.02.2 first but it failed due to https://github.com/mirage/mirage-platform/issues/137
- `opam install mirage`
- Clone mirage-www, build for Xen+DHCP and run
- `sudo reboot` seems to work now (fixes #22)
- SSH dropped once due to corruption. I vaguely remember this being related to TCP offload bugs?
- #23 is not fixed

To be done later (not part of this PR):
- XAPI
- blktap2 (?)
- Check for performance regressions
- Update `README.md` if the changes are considered useful
